### ISSUE TITLE
Normalising reaction time for each player

### ISSUE THREAD

**Issue by scienceanna** [nebogeo/seeme#1]
One of our main concerns is controlling for the difference in capture rate on different platforms – we both found our average times were around a second faster on touch screens than when using a mouse/touchpad, which could mean that a lot of the variance in capture time can be explained by this rather than by anything scientifically more interesting.

One way around this that we thought of could be to standardise the times for each player, so that the overall average time to capture per game is subtracted from the time to capture each target - and then to use these standardised times in the evolutionary algorithm. A simpler alternative might just be to rank the capture times for each individual. Would either of these or something similar be possible?

**Comment by nebogeo** [comment]
Yes this is exactly the thinking we arrived at with egglab but never implemented. We already have the average times for the player, so normalisation for this should be feasible.

**Comment by nebogeo** [comment]
Just revisiting this - I'm a bit worried about putting this in as it needs a lot of testing and we don't have much time. Also it works without it already... however we are now collecting data on this so we can perhaps look at the variance of performance and design this later when we have more information.

**Comment by scienceanna** [comment]
I'm really keen that we do this - even if it requires a lot of testing... I think there will be much bigger between subject than within subject differences and we'll get spurious results if we don't try to control for this. I spoke to Jolyon about egglab and he said that this was a big issue in interpreting the data from that - and I think adding movement is only going to make the differences between playing on different platforms etc. more problematic. Is there any chance you could try implementing it & we can do our best to play lots over the next week or two? If it doesn't seem to be working properly we can always go back to a non normalised version for the real launch, as we know that works at least.

**Comment by nebogeo** [comment]
Ok let's do this then - I had a cup of coffee and shoehorned in what I think we need this afternoon - so we now have:
- Average game scores of players recorded
- A global average of all player scores

However this is going to be quite limited I realise. One problem is that we can't track players over different sessions (for both technical and ethical reasons), so it's more like average scores per session. Also the first game someone plays we can't control for as we need their player average to be calculated first.

To explain this a bit more, there is no way we can go back and apply the control later on, as the pattern fitness needs to be recorded straight away. The reason for this is that there may be many people playing at once, so we need to be able to use the fitness at any time (periodic generation updates can be triggered at any time).

So with this caveat - after the first game someone plays, we have information on their running average - and we can subtract this from the overall player average to give us an idea of their relative performance (greater than 0 is better than average, and the bigger this number the better they are, I think).

So presumably we can add this difference to the fitness recorded (so we add on a bit of time to a patterns fitness for better players, take away a bit of time for worse ones). Does that make sense?

It probably needs a factor to multiply it by - perhaps we add 10% of this difference to the fitness? As each pattern has a fitness calculated by 5 players, each one will be multiplied by this control for each individual player.

**Comment by scienceanna** [comment]
Ohh, okay, I was imagining that the fitness calculations were made at the end of a game, and therefore you could just take an average across the game and subtract this from each RT then divide by the average RT (i.e. Z score), but I now see why this is tricky. Presumably this could be done after the first game, using their running average though? Might be simpler than subtracting from global averages etc. - not really sure how best to biologically "weight" the fitness!

Could the first 5 (or something) targets a player catches not be used for the evolutionary algorithm & just used for average calculating, or is this complicated? (Also, just out of interest, how are you tracking individual players - by IP address?)

**Comment by nebogeo** [comment]
I see... so we are only tracking users by session - if the same person comes back later we don't know. The only way around this without resorting to slightly dodgy practices or cookies (which I'd like to avoid) is having an explicit sign in mechanism - which turns a lot of people off. 

Given that our ID tracking is quite short term I'd worry that continuous Z score with a running average would be more noisy than the effect it's trying to control for. Waiting for a whole game played is much better - but yup we'd lose a lot of input there (as a lot of people only play once). 

I guess my main concern is the biological validity though - a diversity of predator ability seems fairly easy to justify, a pattern that is successful will have to be tested by hundreds of different people with different devices.

Also I'm not sure, but I don't think Jolyon has been looking at the egglab data yet, but the nightjar game which is quite different - much more explicitly about people's perception on the photographs rather than analysing resulting patterns.

I'll have a bit more of a think about this... :)

**Comment by scienceanna** [comment]
Yeah, I agree that cookies/sign in mechanisms aren't the way to go. Okay, in that case, I think I agree with you - if we could normalise across a session I think it would help to reduce noise caused by things we aren't interested in, but if that's not possible, it might be better to leave it completely un-normalised. Shall we leave it as it is for now & I will try to play a bunch of games on different platforms etc. over the next week or so and see what ends up happening? 

Fair point on the nightjar/egglab distinction, I don't think I specifically asked him which he'd been looking at!

**Comment by nebogeo** [comment]
Ok lets see how it goes, I'll leave it how it is...
